### PR TITLE
[UNR-5808] [Draft] Native event tracing 

### PR DESCRIPTION
**Pull request by improbable-danny**
Enable ET in native NFRs and deprecate legacy latency tracer

https://improbableio.atlassian.net/browse/UNR-5808

GDK: https://github.com/spatialos/UnrealGDK/pull/3269
Gyms: https://github.com/spatialos/UnrealGDKTestGyms/pull/456
TestGymBuildKite: https://github.com/improbable/TestGymBuildkite/pull/296
nfr-benchmark-pipeline: https://github.com/improbable/nfr-benchmark-pipeline/pull/184

#### Documentation
[TODO update NFR docs]

### CODE REVIEW

**Review by MatthewSandfordImprobable — APPROVED** [review]

**Comment by MatthewSandfordImprobable** [comment]
Cool stuff. Just so I haven't missed anything... this PR consists of: 
Removing latency tracer code and latency tracer dependencies. 
Reformatting code.
Event Tracers are now passed settings in constructor.

**Comment by improbable-danny** [comment]
>Cool stuff. Just so I haven't missed anything... this PR consists of:
Removing latency tracer code and latency tracer dependencies.
Reformatting code.
Event Tracers are now passed settings in constructor.
I guess your blueprint library will be in another PR or is it customer to testgyms?

Yeah that's right, the test gym PR is here https://github.com/spatialos/UnrealGDKTestGyms/pull/456

File: SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialGDKSpanId.cpp [hunk 1]
```diff
@@ -31,6 +31,23 @@ const Trace_SpanIdType* FSpatialGDKSpanId::GetConstId() const
 	return Id;
 }
 
+FUserSpanId FSpatialGDKSpanId::ToUserSpan(const FSpatialGDKSpanId& SpanId)
+{
+	FUserSpanId UserSpan;
+	UserSpan.Data.SetNum(sizeof(SpanId));
+	memcpy(UserSpan.Data.GetData(), &SpanId, sizeof(FSpatialGDKSpanId));
```
Comment (improbable-danny) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialGDKSpanId.cpp:38]: SetNum with sizeof(SpanId) but memcpy with sizeof(FSpatialGDKSpanId) - they are equal but seems asymmetric! Do we static assert this somewhere btw?

Comment (simonsarginson) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialGDKSpanId.cpp:38]: +1

**Review by simonsarginson — COMMENTED** [review]
One minor comment, otherwise LGTM

**Review by simonsarginson — APPROVED** [review]

**Pull request merged**